### ISSUE TITLE
deleteById does not work

### ISSUE THREAD

**Issue by iMrTidy** [Kodein-Framework/Kodein-DB#44]
I am calling `DB.deleteById<M>(id)` but when I call `DB.getById<M>(id)` I get the object back like it was not deleted. However, if I delete like that:
```
val key = DB.keyById<M>(id)
persistentDB.delete(key)
```
all works as expected. Am I doing something wrong or there is a bug?

**Comment by SalomonBrys** [comment]
Definitely a bug.
Which version are you on?

**Comment by iMrTidy** [comment]
It is 0.8.1-beta

**Comment by Kasheen** [comment]
@iMrTidy When you say `DB.deleteById<M>(key)` are you actually passing a key to that function, or are you passing the id of the model, which is what that function is looking for? Keep in mind a key is not an id.

I wrote the following test and it didn't reproduce the issue on 0.9.0-beta:
```kotlin
@Test
fun test10_getAfterDeleteById() {
    db.inflateDB()

    assertNotNull(db.getById<Adult>("BRYS", "Salomon"))

    db.deleteById<Adult>("BRYS", "Salomon")

    assertNull(db.getById<Adult>("BRYS", "Salomon"))
}
```

**Comment by iMrTidy** [comment]
@Kasheen Sorry, I was not clear in my initial comment which I have edited now. I do use 'id' and not 'key' and it does not work in 0.8.1-beta. It is maybe fixed in the 0.9.0-beta. I will try to find time to check it and report here.

**Comment by jakobkmar** [comment]
Hey, `0.8.1-beta` is not the most recent version - this has been fixed with #38
